packages/gestalt/src/PageHeaderComponents.js

```diff
@@ -0,0 +1,212 @@
+  subtext,
+  helperLink,
+}: {|
+  subtext: string,
+  helperLink?: {|
+    text: string,
+    accessibilityLabel: string,
+    href: string,
+    onClick: ({|
+      event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+      dangerouslyDisableOnNavigation: () => void,
+    |}) => void,
+  |},
+|}): Node {
+  return (
+    <Box display="none" smDisplay="block">
+      <Text inline>
+        {subtext}
+        {helperLink ? (
+          <Fragment>
+            {' '}
+            <Text inline underline>
+              <Link
+                accessibilityLabel={helperLink.accessibilityLabel}
+                href={helperLink.href}
+                onClick={helperLink.onClick}
+                target="blank"
+                inline
+              >
+                <span style={{ textDecoration: 'underline' }}>{helperLink.text}</span>
+              </Link>
+            </Text>
+          </Fragment>
+        ) : null}
+      </Text>
+    </Box>
+  );
+}
+
+export function PageHeaderActionBlock({
+  primaryAction,
+  secondaryAction,
+  dropdownAccessibilityLabel = '',
+}: {|
+  primaryAction: {| component: ActionType, dropdownItems: $ReadOnlyArray<Node> |},
+  secondaryAction?: {| component: ActionType, dropdownItems: $ReadOnlyArray<Node> |},
+  dropdownAccessibilityLabel?: string,
+|}): Node {
+  const [open, setOpen] = useState(false);
+  const anchorRef = useRef(null);
+
+  return (
+    <Fragment>
+      <Box display="none" mdDisplay="block">
+        <Flex gap={2}>
+          {/* 48px height needed to maintain proper sizing when action is a Link */}
+          {secondaryAction ? (
+            <Box height={48} display="flex" alignItems="center">
+              {secondaryAction.component}
+            </Box>
+          ) : null}
+          <Box height={48} display="flex" alignItems="center">
+            {primaryAction.component}
+          </Box>
+        </Flex>
+      </Box>
+      <Box display="block" mdDisplay="none">
+        <IconButton
+          accessibilityControls="pageheader-dropdown"
+          accessibilityExpanded={open}
+          accessibilityHaspopup
+          accessibilityLabel={dropdownAccessibilityLabel}
+          icon="ellipsis"
+          iconColor="darkGray"
+          onClick={() => setOpen((prevVal) => !prevVal)}
+          ref={anchorRef}
+          selected={open}
+          size="lg"
+        />
+        {open && (
+          <Dropdown
+            anchor={anchorRef.current}
+            id="pageheader-dropdown"
+            onDismiss={() => setOpen(false)}
+          >
+            {[...primaryAction.dropdownItems, ...(secondaryAction?.dropdownItems ?? [])]}
+          </Dropdown>
+        )}
+      </Box>
+    </Fragment>
+  );
+}
+
+export function PageHeaderItemsBlock({ items }: {| items: $ReadOnlyArray<Node> |}): Node {
+  return (
+    <Box display="none" lgDisplay="block" overflow="hidden">
+      <Flex gap={2}>
+        {items?.[0] ? items[0] : null}
```

Comment:
This was marked as "done" but there's still unnecessary optional chaining here for the required prop `items`

Reply:
DONE